hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieFileReader.java

```diff
@@ -35,6 +37,14 @@
 
   public Set<String> filterRowKeys(Set<String> candidateRowKeys);
 
+  default Map<String, R> getRecordsByKeys(TreeSet<String> sortedCandidateRowKeys) throws IOException {
```

Comment:
TreeSet will have the keys sorted and its a clue for the HFile scanners to do seekTo the first key and then do the sequential scan instead of repeated full/pointed lookups

Reply:
Sorting keys is only useful for HFileReader. Since this is an interface, it should only impose the common functionality. I feel List<String> is more suited here and HFileReader should internally sort the keys.

Additionally, wherever these keys are being read from will most probably be returning a List<String>. So there is already a step somewhere to create a TreeSet out of a List<> which is an additional copy. So having HFileReader sort internally is not worse in any way and seems more cleaner to me.

Reply:
Fixed.